pkg/proxy/topology_test.go

```diff
@@ -17,376 +17,264 @@ limitations under the License.
 package proxy
 
 import (
-	"reflect"
+	"fmt"
 	"testing"
 
 	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/sets"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	featuregatetesting "k8s.io/component-base/featuregate/testing"
 	"k8s.io/kubernetes/pkg/features"
 )
 
-func TestFilterEndpoints(t *testing.T) {
-	type endpoint struct {
-		ip        string
-		zoneHints sets.String
-		unready   bool
+func checkExpectedEndpoints(expected sets.String, actual []Endpoint) error {
```

Comment:
non-blocking comment: thinking sets.Difference can simplify this function a little bit:
```
var errs []error

actualCopy := sets.NewString()
for _, ep := range actual {
    actualCopy.Insert(ep.String())
}

diff := expected.Difference(actualCopy)
if diff.Len() > 0 {
    return fmt.Errof("unexpected...")
}

return nil
```

Reply:
That catches "expected endpoint wasn't in actual" but it doesn't catch "actual endpoint wasn't in expected". You'd need to check both `expected.Difference(actualCopy)` and `actualCopy.Difference(expected)`, and then at that point it's not really simpler, it's just different.

(ie, `sets` doesn't implement `Xor`)